spec/query_spec.cr

```diff
@@ -91,15 +91,39 @@ describe LuckyRecord::Query do
       count.should eq 1
     end
   end
+
+  describe "#not" do
+    it "negates the given where condition as 'equal'" do
+      insert_a_user(name: "Paul")
+
+      results = UserQuery.new.name.not("not existing").results
+      results.should eq UserQuery.new.results
+
+      results = UserQuery.new.name.not("Paul").results
+      results.should eq [] of User
+
+      insert_a_user(name: "Alex")
+      insert_a_user(name: "Sarah")
+      results = UserQuery.new.name.lower.not("alex").results
+      results.map(&.name).should eq ["Paul", "Sarah"]
+    end
+
+    it "negates any previous condition" do
+      insert_a_user
+
+      results = UserQuery.new.name.not.is("Paul").results
+      results.should eq [] of User
+    end
+  end
 end
```

Comment:
I'd love to see just one more spec like `it "can use not with operators"` that tested `gt` for example. Just testing one is enough to make sure it works. Something along these lines:

```crystal
it "can use not with operators" do
 insert_a_user(age: 28, name: "John")
 insert_a_user(age: 300, name: "Yoda")

  UserQuery.new.age.not.gt(28).results.map(&.name).should eq ["John"]
end
```

Reply:
Done ✅